pkg/app/piped/trigger/determiner.go

```diff
@@ -29,50 +27,81 @@ import (
+	default:
+		return ds.onCommit
+	}
+}
+
+type OnCommandDeterminer struct {
+}
+
+// ShouldTrigger decides whether a given application should be triggered or not.
+func (d *OnCommandDeterminer) ShouldTrigger(_ context.Context, _ *model.Application, appCfg *config.GenericDeploymentSpec) (bool, error) {
+	if appCfg.Trigger.OnCommand.Disabled {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+type OnOutOfSyncDeterminer struct {
+}
+
+// ShouldTrigger decides whether a given application should be triggered or not.
+func (d *OnOutOfSyncDeterminer) ShouldTrigger(_ context.Context, _ *model.Application, appCfg *config.GenericDeploymentSpec) (bool, error) {
+	if appCfg.Trigger.OnOutOfSync.Disabled {
+		return false, nil
+	}
+
+	return true, nil
+}
+
 type LastTriggeredCommitGetter interface {
 	Get(ctx context.Context, applicationID string) (string, error)
 }
 
-type Determiner struct {
+type OnCommitDeterminer struct {
 	repo         git.Repo
 	targetCommit string
 	commitGetter LastTriggeredCommitGetter
-	// Flag `ignoreUserConfig` set to `true` will force check changes and use it to determine
-	// the application deployment should be triggered or not, regardless of the user's configuration.
-	ignoreUserConfig bool
-	logger           *zap.Logger
+	logger       *zap.Logger
 }
 
-func NewDeterminer(repo git.Repo, targetCommit string, cg LastTriggeredCommitGetter, ignoreUserConfig bool, logger *zap.Logger) *Determiner {
-	return &Determiner{
-		repo:             repo,
-		targetCommit:     targetCommit,
-		commitGetter:     cg,
-		ignoreUserConfig: ignoreUserConfig,
-		logger:           logger.Named("determiner"),
+func NewOnCommitDeterminer(repo git.Repo, targetCommit string, cg LastTriggeredCommitGetter, logger *zap.Logger) Determiner {
+	return &OnCommitDeterminer{
+		repo:         repo,
+		targetCommit: targetCommit,
+		commitGetter: cg,
+		logger:       logger.Named("determiner"),
 	}
 }
 
 // ShouldTrigger decides whether a given application should be triggered or not.
-func (d *Determiner) ShouldTrigger(ctx context.Context, app *model.Application) (bool, error) {
+func (d *OnCommitDeterminer) ShouldTrigger(ctx context.Context, app *model.Application, appCfg *config.GenericDeploymentSpec) (bool, error) {
 	logger := d.logger.With(
 		zap.String("app", app.Name),
 		zap.String("app-id", app.Id),
 		zap.String("target-commit", d.targetCommit),
 	)
 
-	// TODO: Add logic to determine trigger or not based on other configuration than onCommit.
-	return d.shouldTriggerOnCommit(ctx, app, logger)
-}
-
-func (d *Determiner) shouldTriggerOnCommit(ctx context.Context, app *model.Application, logger *zap.Logger) (bool, error) {
-	deployConfig, err := loadDeploymentConfiguration(d.repo.GetPath(), app)
-	if err != nil {
-		return false, err
-	}
-
 	// Not trigger in case users disable auto trigger deploy on change and the user config is unignorable.
-	if deployConfig.Trigger.OnCommit.Disabled && !d.ignoreUserConfig {
+	if appCfg.Trigger.OnCommit.Disabled {
```

Comment:
Just to confirm, with this change, the plan-preview feature spec will be changed as: "require to enable trigger.onCommit to enable using plan-preview", right?

Reply:
Yes. I think that is correct because plan-preview is for predicting what will happen after the PR got merged. So disabling `onCommit` means that no deployment will be triggered after.

Reply:
Nice, thanks for your confirmation. Lets update the docs for that after this PR is merged 👍